c++/src/kj/async-inl.h

```diff
@@ -1084,7 +1084,7 @@ inline PromiseForResult<Func, void> evalNow(Func&& func) {
 template <typename Func>
 struct RetryOnDisconnect_ {
   static inline PromiseForResult<Func, void> apply(Func&& func) {
-    auto promise = evalNow(func);
+    auto promise = evalLater(func);
     return promise.catch_([func = kj::mv(func)](kj::Exception&& e) -> PromiseForResult<Func, void> {
```

Comment:
Looks like this may need to be made `mutable` to handle the case where ` func` is mutable... I'll change that after looking into the g++ test failures and get everything working using this internally.

Reply:
We not only needed to make this mutable, but also needed to better handle the lifetime of `func`. We can't blindly pass it to `evalLater` since we were also `kj::mv`ing it into the catch. Fixed in the force push.